src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt

```diff
@@ -30,4 +31,14 @@ class DroneCustomMongoRepository(
 
         mongoTemplate.updateFirst(query, update, Drone::class.java)
     }
+
+    override fun findAllDronesWhereTheRemainingBatteryChargeIsLessThanAndHaveTheStatuses(
+        batteryLevel: Double,
+        statusesList: List<DroneStatus>
+    ): List<Drone> {
+        val query = Query()
+            .addCriteria(Criteria.where("batteryLevel").lte(batteryLevel))
+            .addCriteria(Criteria.where("status").`in`(statusesList))
```

Comment:
I don't force you changing this code, just sharing alternative approach:
```kotlin
val criteria  = Criteria.where("batteryLevel").lte(batteryLevel)
                              .and("status").`in`(statusesList)
val query = Query(criteria)
```

Reply:
Thanks, I was interesting which of the options is better to use?

Reply:
your approach can be used to build some more complex queries. My approach is less verbose but less flexible at the same time